parallel-crypto3/libs/parallel-zk/include/nil/crypto3/zk/snark/systems/plonk/placeholder/gates_argument.hpp

```diff
@@ -108,15 +110,29 @@ namespace nil {
                             math::make_evaluation_domain<FieldType>(extended_domain_size);
 
                         parallel_for(0, variables.size(),
-                            [&variables, &variable_values_out, &assignments, &domain, &extended_domain, extended_domain_size](std::size_t i) {
+                            [&variables, &variable_values_out, &assignments, &domain, &extended_domain, extended_domain_size, &mask_polynomial, &lagrange_0](std::size_t i) {
                                 const variable_type& var = variables[i];
 
                                 // Convert the variable to polynomial_dfs variable type.
                                 polynomial_dfs_variable_type var_dfs(var.index, var.rotation, var.relative,
                                     static_cast<typename polynomial_dfs_variable_type::column_type>(
                                         static_cast<std::uint8_t>(var.type)));
 
-                                polynomial_dfs_type assignment = assignments.get_variable_value(var_dfs, domain);
+                                polynomial_dfs_type assignment;
+                                if( var.index >= 0){

```

Comment:
This code needs to get into function get_variable_value itself. We don't even check here that it's a selector, before returning the special values.

Reply:
Now everywhere we check that var.type == selector